app/code/Magento/GiftMessageGraphQl/Model/Resolver/Cart/GiftMessage.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GiftMessageGraphQl\Model\Resolver\Cart;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\GiftMessage\Api\CartRepositoryInterface;
+
+/**
+ * Class provides ability to get GiftMessage for cart
+ */
+class GiftMessage implements ResolverInterface
+{
+    /**
+     * @var CartRepositoryInterface
+     */
+    private $cartRepository;
+
+    /**
+     * GiftMessage constructor.
+     *
+     * @param CartRepositoryInterface $cartRepository
+     */
+    public function __construct(
+        CartRepositoryInterface $cartRepository
+    ) {
+        $this->cartRepository = $cartRepository;
+    }
+
+    /**
+     * Return information about Gift message of cart
+     *
+     * @param Field            $field
+     * @param ContextInterface $context
+     * @param ResolveInfo      $info
+     * @param array|null       $value
+     * @param array|null       $args
+     *
+     * @return array|Value|mixed
+     *
+     * @throws GraphQlInputException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($value['model'])) {
+            throw new GraphQlInputException(__('"model" value should be specified'));
+        }
+        $cart = $value['model'];
+
+        try {
+            $giftCartMessage = $this->cartRepository->get($cart->getId());
+        } catch (LocalizedException $e) {
+            throw new GraphQlInputException(__('Can\'t load cart.'));
+        }
+
+        return [
```

Comment:
I believe it is better to return null if the cart does not have a gift message, rather than returning an object with empty fields. This would align more with schema as you have specified theses fields as non-nullable.

Reply:
**Done**